rust nm: Improve performance for massive apply

cathay4t opened the pull request:
Changed the code to only extend checkpoint timeout when half of it
elapsed. Also changed the default timeout for checkpoint to 60 seconds
to align with python code.

Also skip reapply is there is no active connection.

Also skip activation for new controller and new ports as controller
activation will automatically activate its new ports.

Warning user if interface count exceeded 1000 in single desire state.

This is performance result on the same host:

    100 bridge over 100 vlan
        Rust:       5.728s
        Python:     6.423s

    300 bridge over 300 vlan
        Rust:       34.571s
        Python:     38.903s

    500 bridge over 500 vlan
        Rust:       1m39.321s
        Python:     1m40.501s

    1000 bridge over 1000 vlan
        Rust:       8m33.536s
        Python:     5m57.810s

Considering 1000+ interfaces are not supported, this result is
acceptable.

Enabled slow integration test cases for rust.

cathay4t commented:
The rust code is spending more time(20 seconds) than python code when creating 1000 vlans and 1000 bridges. Still investigating.

cathay4t commented:
I have tried all the steps I could find out, still cannot improve performance of 2000 interfaces.
Considering rust code is doing better at 1000 interfaces level which is also the maximum supported interface count, I would like to move on with this approach.

ffmancera reviewed (changes requested):
There are failures related to bridge/ports relation.. could you check the rust CI? Thanks!

cathay4t committed:
rust nm: Improve performance for massive apply

Changed the code to only extend checkpoint timeout when half of it
elapsed. Also changed the default timeout for checkpoint to 60 seconds
to align with python code.

Also skip reapply is there is no active connection.

Also skip activation for new controller and new ports as controller
activation will automatically activate its new ports.

Warning user if interface count exceeded 1000 in single desire state.

This is performance result on the same host:

    100 bridge over 100 vlan
        Rust:       5.728s
        Python:     6.423s

    300 bridge over 300 vlan
        Rust:       34.571s
        Python:     38.903s

    500 bridge over 500 vlan
        Rust:       1m39.321s
        Python:     1m40.501s

    1000 bridge over 1000 vlan
        Rust:       8m33.536s
        Python:     5m57.810s

Considering 1000+ interfaces are not supported, this result is
acceptable.

Enabled slow integration test cases for rust.

Signed-off-by: Gris Ge <fge@redhat.com>

cathay4t commented:
> There are failures related to bridge/ports relation.. could you check the rust CI? Thanks!

OVS port does not have auto-port activation. Fixed.

cathay4t committed:
ovs test: remove xfail test_ovs_service_missing_with_system_port_only

The old problem is fixed in NetworkManager-1.37.2, removing xfail.

Signed-off-by: Gris Ge <fge@redhat.com>

cathay4t committed:
test: xfail `test_external_managed_subordnates`

It is known bug that `test_external_managed_subordnates` might fail:

https://bugzilla.redhat.com/show_bug.cgi?id=2070855

Signed-off-by: Gris Ge <fge@redhat.com>
